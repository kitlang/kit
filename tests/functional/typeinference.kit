function voidFunc1() {
    return;
}
function voidFunc2() {
    printf("hello!\n");
}
function voidFunc3() {}
function voidFunc4(): Void {}

function numFunc1() {
    return 1;
}

function numFunc2() {
    return 1 + 2;
}

function numFunc3(): Float {
    return numFunc1();
}

function main() {
    voidFunc1();
    voidFunc2();
    voidFunc3();
    voidFunc4();
    var a = numFunc1();
    var b = numFunc2();
    var c = numFunc3();

    printf("%.2f\n", a + b + c);
}
