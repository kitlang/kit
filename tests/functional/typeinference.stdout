hello!
5.00
